File: offer03.cpp
class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
        int i = 0;
        while(i<nums.size()){
            if(nums[i] == i) {
                i++;
                continue;
            }
            if(nums[nums[i]]==nums[i]){
                return nums[i];
            }
            else{
                swap(nums[i],nums[nums[i]]);
            }
        }
        return -1;
    }
};

File: offer03.py
class Solution:
    def findRepeatNumber(self, nums: List[int]) -> int:
        i: int = 0
        while(i<len(nums)):
            if nums[i] == i:
                i+=1
                continue
            if nums[i]==nums[nums[i]]:
                return nums[i]
            else:
                nums[nums[i]],nums[i] = nums[i],nums[nums[i]]
                
        return -1
